### PR TITLE
fix: Ignore iceberg columns that are not current

### PR DESCRIPTION
**Pull request by nicor88**
### Description
Iceberg support schema evolution, therefore columns can be dropped and added.
If we drop a column the column is somehow returned by the glue api, but it has an atrribute that can be used to pick only current columns. 
This PR aim to fix this issue.

### Iceberg append

model: iceberg_testing

```
{{ config(
  materialized='incremental',
  incremental_strategy='append',
  table_type='iceberg',
) }}

SELECT
   'example'  as name,
   1 as user_id,
  cast(replace(cast(at_timezone(current_timestamp, 'Europe/Berlin') as varchar), ' Europe/Berlin', '') as timestamp(6)) as inserted_at
```

* then go run this `ALTER TABLE iceberg_testing DROP COLUMN inserted_at`
* then run the new model removing inserted_at

```
{{ config(
  materialized='incremental',
  incremental_strategy='append',
  table_type='iceberg',
) }}

SELECT
   'example'  as name,
   1 as user_id
```
The model should run because inserted_at=`'iceberg.field.current'='false'` and the field is excluded from the insert statement.

### Iceberg merge

model: iceberg_testing

```
{{ config(
  materialized='incremental',
  incremental_strategy='merge',
  table_type='iceberg',
) }}

SELECT
   'example'  as name,
   1 as user_id,
  cast(replace(cast(at_timezone(current_timestamp, 'Europe/Berlin') as varchar), ' Europe/Berlin', '') as timestamp(6)) as inserted_at
```

* then go run this `ALTER TABLE iceberg_testing DROP COLUMN inserted_at`
* then run the new model removing inserted_at

```
{{ config(
  materialized='incremental',
  incremental_strategy='merge',
  table_type='iceberg',
) }}

SELECT
   'example_2'  as name,
   2 as user_id
```
The model should run because inserted_at=`'iceberg.field.current'='false'` and the field is excluded from the insert statement.

## Checklist
- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

File: dbt/adapters/athena/impl.py [hunk 1]
```diff
@@ -631,6 +631,18 @@ def list_schemas(self, database: str) -> List[str]:
...
+    def _is_current_column(col: dict) -> bool:
+        """
+        Check if a column is current, it always fallback to True if the parameter is not property is not available
+        """
+        if col.get("Parameters", {}).get("iceberg.field.current") == "true":
+            return True
+        elif col.get("Parameters", {}).get("iceberg.field.current") == "false":
+            return False
+        else:
+            return True
```
Comment (svdimchenko) [inline dbt/adapters/athena/impl.py:644]:
```suggestion
        if col.get("Parameters", {}).get("iceberg.field.current") == "false":
            return False
        return True
```

Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:644]: does it make sense to simplify the logic a bit ?

File: dbt/adapters/athena/impl.py [hunk 2]
```diff
@@ -640,6 +652,10 @@ def get_columns_in_relation(self, relation: AthenaRelation) -> List[Column]:
             glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
 
         table = glue_client.get_table(DatabaseName=relation.schema, Name=relation.identifier)["Table"]
-        return [
-            Column(c["Name"], c["Type"]) for c in table["StorageDescriptor"]["Columns"] + table.get("PartitionKeys", [])
-        ]
+
+        columns = [c for c in table["StorageDescriptor"]["Columns"] if self._is_current_column(c) is True]
```
Comment (svdimchenko) [inline dbt/adapters/athena/impl.py:656]:
```suggestion
        columns = [c for c in table["StorageDescriptor"]["Columns"] if self._is_current_column(c)]
```
seems to redundant check here

File: dbt/adapters/athena/impl.py (hunk 1 above)
Comment (nicor88) [reply dbt/adapters/athena/impl.py:644]: We can, I just wanted to be "explicit", but your suggestion is more elegant.

File: dbt/adapters/athena/impl.py (hunk 2 above)
Comment (nicor88) [reply dbt/adapters/athena/impl.py]: indeed, good point.

**Comment by svdimchenko** [comment]
great job done! code looks good to me

**Pull request merged**